src/main/java/org/blockchain_innovation/factom/identiy/did/request/CreateFactomDidRequest.java

```diff
@@ -0,0 +1,162 @@
+package org.blockchain_innovation.factom.identiy.did.request;
+
+import org.blockchain_innovation.factom.identiy.did.DIDVersion;
+import org.blockchain_innovation.factom.identiy.did.entry.CreateFactomDIDEntry;
+import org.factomprotocol.identity.did.model.DidMethodVersion;
+import org.factomprotocol.identity.did.model.FactomDidContent;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CreateFactomDidRequest {
+    private final DIDVersion didVersion;
+    private final List<CreateKeyRequest> managementKeys;
+    private final List<CreateKeyRequest> didKeys;
+    private final List<CreateServiceRequest> services;
+    private final String[] tags;
+    private final String nonce;
+    private final String networkName;
+
+    private CreateFactomDidRequest(DIDVersion didVersion, String networkName, List<CreateKeyRequest> managementKeys, List<CreateKeyRequest> didKey, List<CreateServiceRequest> service, String nonce, String... tags) {
+        this.didVersion = didVersion;
+        this.networkName = networkName;
+        this.managementKeys = managementKeys;
+        this.didKeys = didKey;
+        this.services = service;
+        this.nonce = nonce;
+        this.tags = tags;
+    }
+
+    public String getNonce() {
+        return this.nonce;
+    }
+
+    public String[] getTags() {
+        return this.tags;
+    }
+
+    public DIDVersion getDidVersion() {
+        return this.didVersion;
+    }
+
+    public FactomDidContent toFactomDidContent() {
+        String chainId = new CreateFactomDIDEntry(this.didVersion, null, this.nonce, this.tags).getChainId();
+        String did = this.networkName == null ? "did:factom:" + chainId : "did:factom:" + this.networkName + ':' + chainId;
```

Comment:
Maybe also check whether we cannot get something like "mainnet" for the network name, in which case the network name should not become part of the did

Reply:
updated.

Reply:
Made it case insensitve